src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/DeductionDateMapper.java

```diff
@@ -0,0 +1,24 @@
+package uk.nhs.digital.nhsconnect.nhais.mapper;
+
+import org.hl7.fhir.r4.model.Parameters;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.DateTimePeriod;
+import uk.nhs.digital.nhsconnect.nhais.model.fhir.ParameterNames;
+import uk.nhs.digital.nhsconnect.nhais.model.fhir.ParametersExtension;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+
+@Component
+public class DeductionDateMapper implements FromFhirToEdifactMapper<DateTimePeriod> {
+
+    public DateTimePeriod map(Parameters parameters) {
+        return new DateTimePeriod(getDeductionDate(parameters), DateTimePeriod.TypeAndFormat.DEDUCTION_DATE);
+    }
+
+    private Instant getDeductionDate(Parameters parameters) {
+        String dateAsString = ParametersExtension.extractValue(parameters, ParameterNames.DATE_OF_DEDUCTION);
+        return LocalDate.parse(dateAsString).atStartOfDay(ZoneOffset.UTC).toInstant();
```

Comment:
Date of entry/exit uses:
`return localDate.atStartOfDay(TimestampService.UKZone).toInstant();`
which one is right?

Reply:
good question I just copied this from acceptance. Since they are date it might be better to use LocalDate all the way through instead of assigning them a nominal time.

Reply:
So `return Instant.from(localDate);` ?

Reply:
I think you can't get `Instant` without time part